product_label_image/reports/product_template_report.xml

```diff
@@ -0,0 +1,66 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<odoo>
+    <data>
+        <template id="report_simple_barcode_with_image" inherit_id="product.report_simple_label2x7">
+            <xpath expr="//td" position="attributes">
+                <attribute name="style">{'display': inline-block;}</attribute>
+            </xpath>
+            <xpath expr="//div" position="after">
+                <t t-if="product.image_512">
+                    <div style="width: 97mm; border:1px solid #555555; height:37.1mm;" class="o_label_full">
+                        <div class="text-center align-middle" style="height: 10rem;">
+                            <img t-att-src="'data:image/png;base64,%s' % to_text(product.image_512)"
+                                 style="height:109px;width:auto;"/>
+                        </div>
+                    </div>
+                </t>
+            </xpath>
```

Comment:
In the base module inside the template "report_simple_label2x7" and other inherited modules in this file, you can find the core functionality to display the Barcode. So, may be we can use that?

Reply:
I have just updated code and added new option 4x6. So removed old template inheritance and created new fresh report for 4x6 size.